heat/core/arithmetics.py

```diff
@@ -197,6 +203,179 @@ def bitwise_xor(t1, t2):
     return operations.__binary_op(torch.Tensor.__xor__, t1, t2)
 
 
+def cumprod(a, axis, dtype=None, out=None):
+    """
+    Return the cumulative product of elements along a given axis.
+
+    Parameters
+    ----------
+    a : DNDarray
+        Input array.
+    axis : int
+        Axis along which the cumulative product is computed.
+    dtype : dtype, optional
+        Type of the returned array, as well as of the accumulator in which
+        the elements are multiplied.  If *dtype* is not specified, it
+        defaults to the dtype of `a`, unless `a` has an integer dtype with
+        a precision less than that of the default platform integer.  In
+        that case, the default platform integer is used instead.
+    out : DNDarray, optional
+        Alternative output array in which to place the result. It must
+        have the same shape and buffer length as the expected output
+        but the type of the resulting values will be cast if necessary.
+
+    Returns
+    -------
+    cumprod : DNDarray
+        A new array holding the result is returned unless `out` is
+        specified, in which case a reference to out is returned.
+    """
```

Comment:
Add Raises section for exceptions that are thrown from within this function

Reply:
Shall the raises section only be in  __cum_op() or also here?